app/components/scheduled_post_indicator/scheduled_post_indicator.test.tsx

```diff
@@ -0,0 +1,156 @@
+        const {getByText} = renderWithEverything(
+            <ScheduledPostIndicator
+                isThread={true}
+                scheduledPostCount={10}
+            />,
+            {database},
+        );
+
+        expect(getByText(/10 scheduled messages in thread./)).toBeVisible();
+        expect(getByText(/See all./)).toBeVisible();
+    });
+
+    it('renders with military time when preference is set', async () => {
+        await operator.handlePreferences({
+            preferences: [
+                {
+                    user_id: 'user_1',
+                    category: 'display_settings',
+                    name: 'use_military_time',
+                    value: 'true',
+                },
+            ],
+            prepareRecordsOnly: false,
+        });
+
+        const {getByText, findByTestId} = renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeVisible();
+
+        expect(getByText(/19:41/)).toBeVisible();
+    });
+
+    it('renders with 12-hour time when preference is set to 12 hours', async () => {
+        await operator.handlePreferences({
+            preferences: [
+                {
+                    user_id: 'user_1',
+                    category: 'display_settings',
+                    name: 'use_military_time',
+                    value: 'false',
+                },
+            ],
+            prepareRecordsOnly: false,
+        });
+
+        const {getByText, findByTestId} = renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeVisible();
+
+        expect(getByText(/7:41 PM/)).toBeVisible();
+    });
+
+    it('renders with 12-hour time when preference is not set', async () => {
+        const {getByText, findByTestId} = renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeVisible();
+
+        expect(getByText(/7:41 PM/)).toBeVisible();
+    });
+
+    it('handles missing current user', async () => {
+        const {getByText, findByTestId} = renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeVisible();
+        expect(getByText(/Message scheduled for/)).toBeVisible();
+    });
```

Comment:
There are a couple of prop that can be undefined that you are not testing: scheduledPostCount and isThread. `isThread` may not be needed, since undefined is falsy, and you are setting the default value for scheduledPostCount to 0, but I guess it wouldn't hurt to add test to make sure those defaults never change.

Reply:
`scheduledPostCount` will default to 0 once I integrate it with local database and will never be undefined.